instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java

```diff
@@ -56,22 +57,19 @@
   private final SpanStatusExtractor<? super REQUEST, ? super RESPONSE> spanStatusExtractor;
   private final List<? extends AttributesExtractor<? super REQUEST, ? super RESPONSE>> extractors;
   private final ErrorCauseExtractor errorCauseExtractor;
-
-  Instrumenter(
-      String instrumentationName,
-      Tracer tracer,
-      SpanNameExtractor<? super REQUEST> spanNameExtractor,
-      SpanKindExtractor<? super REQUEST> spanKindExtractor,
-      SpanStatusExtractor<? super REQUEST, ? super RESPONSE> spanStatusExtractor,
-      List<? extends AttributesExtractor<? super REQUEST, ? super RESPONSE>> extractors,
-      ErrorCauseExtractor errorCauseExtractor) {
-    this.instrumentationName = instrumentationName;
-    this.tracer = tracer;
-    this.spanNameExtractor = spanNameExtractor;
-    this.spanKindExtractor = spanKindExtractor;
-    this.spanStatusExtractor = spanStatusExtractor;
-    this.extractors = extractors;
-    this.errorCauseExtractor = errorCauseExtractor;
+  private final StartTimeExtractor<REQUEST> startTimeExtractor;
+  private final EndTimeExtractor<RESPONSE> endTimeExtractor;
+
+  Instrumenter(InstrumenterBuilder<REQUEST, RESPONSE> builder) {
+    this.instrumentationName = builder.instrumentationName;
+    this.tracer = builder.openTelemetry.getTracer(builder.instrumentationName);
+    this.spanNameExtractor = builder.spanNameExtractor;
+    this.spanKindExtractor = builder.spanKindExtractor;
+    this.spanStatusExtractor = builder.spanStatusExtractor;
+    this.extractors = new ArrayList<>(builder.attributesExtractors);
+    this.errorCauseExtractor = builder.errorCauseExtractor;
+    this.startTimeExtractor = builder.startTimeExtractor;
+    this.endTimeExtractor = builder.endTimeExtractor;
   }
```

Comment:
I refactored the Instrumenter constructor and used the same approach as OkHttp builder. I think it makes passing all those extractors a bit easier on the eyes